types/result.go

```diff
@@ -39,19 +39,42 @@ func (res Result) IsOK() bool {
 	return res.Code.IsOK()
 }
 
-// Is a version of TxResponse where the tags are StringTags rather than []byte tags
+// IndexedABCILogs represents a slice of IndexedABCILog.
+type IndexedABCILogs []IndexedABCILog
+
+// IndexedABCILog defines a structure containing an indexed tx ABCI message log.
+type IndexedABCILog struct {
+	MsgIndex int    `json:"msg_index"`
+	Success  bool   `json:"success"`
+	Log      string `json:"log"`
+}
+
+// String implements the fmt.Stringer interface for the IndexedABCILogs type.
+func (logs IndexedABCILogs) String() (str string) {
+	if logs != nil {
+		raw, err := json.Marshal(logs)
+		if err == nil {
+			str = string(raw)
+		}
+	}
+
+	return str
+}
+
+// TxResponse defines a structure containing relevant tx data and metadata. The
+// tags are stringified and the log is JSON decoded.
 type TxResponse struct {
-	Height    int64      `json:"height"`
-	TxHash    string     `json:"txhash"`
-	Code      uint32     `json:"code,omitempty"`
-	Data      []byte     `json:"data,omitempty"`
-	Log       string     `json:"log,omitempty"`
-	Info      string     `json:"info,omitempty"`
-	GasWanted int64      `json:"gas_wanted,omitempty"`
-	GasUsed   int64      `json:"gas_used,omitempty"`
-	Tags      StringTags `json:"tags,omitempty"`
-	Codespace string     `json:"codespace,omitempty"`
-	Tx        Tx         `json:"tx,omitempty"`
+	Height    int64           `json:"height"`
+	TxHash    string          `json:"txhash"`
+	Code      uint32          `json:"code,omitempty"`
+	Data      []byte          `json:"data,omitempty"`
+	Logs      IndexedABCILogs `json:"logs,omitempty"`
```

Comment:
s/Logs/LogEntries/ maybe? LogLines? `Entries` sounds better IMHO

Reply:
Hmmm, I don't follow. They are logs, so what is wrong with `Logs` as a field?